client/lib/feed-post-store/index.js

```diff
@@ -198,7 +198,7 @@ function receivePostFromPage( newPost ) {
 }
 
 function receivePost( feedId, postId, post ) {
-	if ( post.errors ) {
+	if ( post.errors || post.status_code === 404 ) {
```

Comment:
@lamosty as long as we're here... can you take a peek at everyone who calls into here and make sure we're never passing null? Or add a `! post` check and bail if the post is null? Right now a null post would blow up this method and throw.